lxd/network/network_utils.go

```diff
@@ -639,6 +639,111 @@ func randomSubnetV6() (string, error) {
 	return "", fmt.Errorf("Failed to automatically find an unused IPv6 subnet, manual configuration required")
 }
 
+// noAvailableAddressErr is used by randomAddressInSubnet to indicate that the subnet was exhausted while searching for
+// IP addresses.
+type noAvailableAddressErr struct {
+	error
+}
+
+// Unwrap implements xerrors.Unwrap for noAvailableAddressErr.
+func (e noAvailableAddressErr) Unwrap() error {
+	return e.error
+}
+
+// randomAddressInSubnet finds a random address within a given subnet. If given, the validate function is called for
+// each randomly generated value. Each time validate returns an error, another IP address will be generated until all
+// IPs have been exhausted or the context is cancelled.
+func randomAddressInSubnet(ctx context.Context, ipnet net.IPNet, validate func(net.IP) error) (net.IP, error) {
+	ones, size := ipnet.Mask.Size()
+	subnetExponent := size - ones
+
+	// networkAddress is the first address in the subnet.
+	var networkAddress net.IP
+	if ipnet.IP.To4() != nil {
```

Comment:
Lets do this just once and store in a var, as its called a number of numbers in this function unnecessarily.